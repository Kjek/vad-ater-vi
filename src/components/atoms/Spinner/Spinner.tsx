import { cn } from '@util/cn';
import type { HTMLAttributes, SVGProps } from 'react';

interface SpinnerProps extends HTMLAttributes<HTMLDivElement> {
  svgProps?: SVGProps<SVGElement>;
}

const Spinner = (props: SpinnerProps) => {
  return (
    <div {...props} className={cn('flex self-center', props.className)}>
      <svg
        className='my-auto h-5 w-5 animate-spin text-gray-800 dark:text-gray-300'
        xmlns='http://www.w3.org/2000/svg'
        fill='none'
        viewBox='0 0 24 24'
      >
        <path
          className='opacity-75'
          fill='currentColor'
          d='M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z'
        ></path>
      </svg>
    </div>
  );
};

export default Spinner;
