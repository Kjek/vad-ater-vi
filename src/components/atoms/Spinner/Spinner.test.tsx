import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import Spinner from './Spinner';

describe(Spinner, () => {
  it('renders correctly', () => {
    const { getByTestId } = render(<Spinner data-testid={'spinner'} />);
    const spinner = getByTestId('spinner');
    expect(spinner).toBeInTheDocument();
    expect(spinner).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(<Spinner />);
    expect(defaultRender).toMatchSnapshot();
  });
});
