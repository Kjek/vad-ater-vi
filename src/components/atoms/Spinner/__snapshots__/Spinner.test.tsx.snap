// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex self-center"
      >
        <svg
          class="my-auto h-5 w-5 animate-spin text-gray-800 dark:text-gray-300"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="opacity-75"
            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex self-center"
    >
      <svg
        class="my-auto h-5 w-5 animate-spin text-gray-800 dark:text-gray-300"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="opacity-75"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
